backend/core/src/main/java/org/sonarsource/sonarlint/core/ConnectionManager.java

```diff
@@ -145,24 +148,26 @@ private HttpClient getClientFor(EndpointParams params, Either<TokenDto, Username
    * Throws ResponseErrorException if connection with provided ID is not found in ConnectionConfigurationRepository
    */
   public ServerConnection getConnectionOrThrow(String connectionId) {
-    var serverApi = getServerApiOrThrow(connectionId);
-    return new ServerConnection(connectionId, serverApi, client);
+    return connectionCache.computeIfAbsent(connectionId, connId -> {
+      var serverApi = getServerApiOrThrow(connId);
+      return new ServerConnection(connId, serverApi, client);
+    });
   }
 
   /**
    * Returns empty Optional if connection with provided ID is not found in ConnectionConfigurationRepository
    */
   public Optional<ServerConnection> tryGetConnection(String connectionId) {
```

Comment:
Not directly related to your change, but as we discussed before the API surface for this manager is huge. I think this method could be removed. It's used in a single place, where I guess we could use getConnectionOrThrow

Reply:
Unfortunately, it's not a trivial change. I will address it in a separate PR later.